hyper/hyper/hyper.go

```diff
@@ -10,25 +10,38 @@ import (
 
 // Interface is an interface with all the actions that a hypervisor has to be able to do
 type Interface interface {
-	// Get returns a desktop using it's name
+	// Get returns a running desktop object from it's name
 	Get(name string) (*libvirt.Domain, error)
 
-	// Start starts a new machine using the provided XML
+	// Start starts a new machine using the provided XML definition
+	// so it is a non-persistent desktop from libvirt point of view
 	Start(xml string, options *StartOptions) (*libvirt.Domain, error)
 
 	// Stop stops a running desktop
 	Stop(desktop *libvirt.Domain) error
 
-	// XMLGet returns the XML definition of a desktop
+	// Suspend suspends a running desktop temporarily saving its memory state. It won't persist hypervisor restarts.
+	Suspend(desktop *libvirt.Domain) error
+
+	// Resume resumes a suspended desktop to its original running state, continuingly execution where it left off.
+	Resume(desktop *libvirt.Domain) error
+
+	// Save saves a running desktop saving its memory state to a file. Will persiste hypervisor restart.
+	Save(desktop *libvirt.Domain, path string) error
+
+	// Restore restores a saved desktop to its original running state, continuing execution where it left off.
+	Restore(path string) error
+
+	// XMLGet returns the running XML definition of a running desktop, so it will have running parameters.
```

Comment:
"XMLGet returns the running XML definition of a running desktop" That's enough

Reply:
ok